lib/scenic/view.rb

```diff
@@ -42,13 +47,14 @@ def ==(other)
 
     # @api private
     def to_schema
-      materialized_option = materialized ? "materialized: true, " : ""
+      materialized_option = materialized || materialized_no_data ? "materialized: true, " : ""
```

Comment:
Here I think we'd only want to be materialized if `materialized` was set. This is not a public interface really so I think we can be a bit more strict. If you want materialized and no data, you'd have to pass both. If you pass materialized: false and no_data: true, we can either ignore no_data or raise.